### PR TITLE
feat: Add quantum reservoir hybrid brain

### PR DESCRIPTION
**Pull request by chrisjz**
Changes:

- Add Quantum Reservoir Hybrid (QRH) brain.
- Add Classical Reservoir Hybrid (CRH) brain.
- Add shared quantum reservoir library.
- Add shared quantum utils library.
- Use extracted shared functions in QRC brain.
- Add distance scaled reward/penalty for predator evasion.
- Add distance scaled reward/penalty for dangerous temperature avoidance.
- Add MI (Mutual Information) analysis script to evaluate the structured topology for QRH.
- Add `scikit-learn` dependency for MI analysis.
- Update documentation related to new brain additions.

## Summary by CodeRabbit

* **New Features**
  * Added two new brain types (QRH quantum hybrid, CRH classical hybrid) and many example experiment configs (foraging, thermotaxis, pursuit predators).
  * Added a mutual-information analysis script and sample MI result artifacts.
  * Reward enhancements: distance-scaled predator evasion and temperature-avoidance rewards plus configurable temperature-proximity penalty.

* **Documentation**
  * Updated README, CONTRIBUTING, and AGENTS docs/specs to describe QRH/CRH and new configs.

* **Tests**
  * Large suite of unit/integration tests added for new brains, utilities, and training flows.

### CODE REVIEW

File: packages/quantum-nematode/quantumnematode/brain/arch/_quantum_utils.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    result = job.result()
+    counts = result.get_counts()
+
+    num_states = 2**num_qubits
+    probs = np.zeros(num_states)
+
+    for bitstring, count in counts.items():
+        idx = int(bitstring, 2)
+        probs[idx] = count / shots
+
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply packages/quantum-nematode/quantumnematode/brain/arch/_quantum_utils.py]: Unlike the `readout_type` case (which silently gave wrong behavior), dividing by `shots=0` would raise a `ZeroDivisionError` immediately — it doesn't silently produce wrong results. And `shots=-1` would be caught by Qiskit's `backend.run()` before we even reach the division.

Adding a guard here is unnecessary — the failure mode is already loud, not silent.

File: packages/quantum-nematode/quantumnematode/brain/arch/_reservoir_hybrid_base.py [hunk 2]
```diff
@@ -0,0 +1,811 @@
...
+    # Learning
+    # =========================================================================
+
+    def learn(
+        self,
+        params: BrainParams,  # noqa: ARG002
+        reward: float,
+        *,
+        episode_done: bool = False,
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply packages/quantum-nematode/quantumnematode/brain/arch/_reservoir_hybrid_base.py]: Good catch — this is a real off-by-one in the GAE bootstrap. You're right that `self.last_value` is V(s_t) (set during `run_brain()` at line 536) rather than V(s_{t+1}), so non-terminal buffer flushes bootstrap with the wrong value.

Impact is bounded in practice:

- Only affects mid-episode buffer boundaries (1 out of every 128 transitions). Terminal flushes are unaffected since `next_non_terminal` zeros out the bootstrap.
- V(s_t) and V(s_{t+1}) are typically close for adjacent states in our small-grid environments with gradual sensory changes, so the absolute error is small.
- The error propagates through GAE with exponential decay (γλ ≈ 0.95), so it mainly biases advantages for the last few transitions in each rollout segment.
- The bias is systematic but consistent across both QRH and CRH (shared base class), so it doesn't affect our ablation comparisons.

Deferring the fix to a follow-up to avoid changing training dynamics mid-ablation study. Filed as a [known issue](https://github.com/SyntheticBrains/nematode/issues/62). Thanks for the sharp eye.

File: openspec/changes/archive/2026-02-26-add-quantum-reservoir-hybrid-brain/specs/qrh-brain/spec.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+The QRHBrain SHALL use statevector simulation for exact feature computation.
+
+#### Scenario: Statevector Execution
+
+- **WHEN** the reservoir circuit is executed for feature extraction
+- **THEN** the system SHALL use Qiskit Statevector simulation (not shot-based measurement)
+- **AND** SHALL compute exact probability amplitudes without shot noise
+- **AND** SHALL NOT add measurement gates to the circuit
+
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply openspec/changes/archive/2026-02-26-add-quantum-reservoir-hybrid-brain/specs/qrh-brain/spec.md]: This should be resolved in the last push, please verify.

File: openspec/changes/archive/2026-02-26-add-quantum-reservoir-hybrid-brain/design.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+- Input encoding helpers shared between QRC and QRH
+
+**Guard**: Only extract if QRC's existing tests continue to pass without modification. If coupling is too tight, accept duplication and note for future cleanup.
+
+### Decision 7: Readout Architecture
+
+**Choice**: Two separate MLPs — actor (features → action logits) and critic (features → value scalar), both 2-layer with 64 hidden units, orthogonal initialization
+
+**Rationale**: Standard PPO actor-critic pattern. Using the shared `build_readout_network()` for both. Separate optimizers allow independent learning rates if needed.
+
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply openspec/changes/archive/2026-02-26-add-quantum-reservoir-hybrid-brain/design.md]: This should be resolved in the last push, please verify.
